flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -299,9 +374,73 @@ public void resetToCheckpoint(long checkpointId, @Nullable byte[] checkpointData
     }
 
     private void checkpointCoordinatorInternal(
-            final long checkpointId, final CompletableFuture<byte[]> result) {
+            long checkpointId, CompletableFuture<byte[]> result) {
         mainThreadExecutor.assertRunningInMainThread();
 
+        try {
+            if (currentPendingCheckpointId != OperatorCoordinator.NO_CHECKPOINT
```

Comment:
Would it be simpler to use `if (currentPendingCheckpointId != OperatorCoordinator.NO_CHECKPOINT)` here?